File: Java/src/main/java/parrot/Parrot.java
package parrot;

public abstract class Parrot {

    private int numberOfCoconuts;
    private double voltage;

    public Parrot(int numberOfCoconuts, double voltage, boolean isNailed) {
    }

    public double getSpeed() {
        throw new RuntimeException("Should be unreachable");
    }

    public double getBaseSpeed() {
        return 12.0;
    }

}

File: TypeScript/jest/src/parrot.ts
export abstract class Parrot {
    protected readonly numberOfCoconuts: number;
    protected readonly voltage: number;
    protected readonly isNailed: boolean;

    protected constructor( numberOfCoconuts: number, voltage: number, isNailed: boolean) {
        this.numberOfCoconuts = numberOfCoconuts;
        this.voltage = voltage;
        this.isNailed = isNailed;
    }

    public abstract getSpeed(): number

    protected getBaseSpeed(): number {
        return 12;
    }
}
